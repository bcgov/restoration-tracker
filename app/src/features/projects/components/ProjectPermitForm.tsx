import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import InputLabel from '@material-ui/core/InputLabel';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { mdiPlus, mdiTrashCanOutline } from '@mdi/js';
import Icon from '@mdi/react';
import CustomTextField from 'components/fields/CustomTextField';
import { FieldArray, useFormikContext } from 'formik';
import React from 'react';
import yup from 'utils/YupSchema';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    formButtons: {
      '& button': {
        margin: theme.spacing(0.5)
      }
    },
    addRowButton: {
      fontWeight: 700
    },
    customListItem: {
      paddingTop: 0,
      paddingBottom: 0,
      paddingLeft: 0,
      paddingRight: theme.spacing(8)
    },
    input: {
      display: 'none'
    }
  })
);

export interface IProjectPermitFormArrayItem {
  permit_number: string;
  permit_type: string;
}

export interface IProjectPermitForm {
  permit: {
    permits: IProjectPermitFormArrayItem[];
  };
}

export const ProjectPermitFormArrayItemInitialValues: IProjectPermitFormArrayItem = {
  permit_number: '',
  permit_type: ''
};

export const ProjectPermitFormInitialValues: IProjectPermitForm = {
  permit: {
    permits: [ProjectPermitFormArrayItemInitialValues]
  }
};

export const ProjectPermitFormYupSchema = yup.object().shape({
  permit: yup.object().shape({
    permits: yup
      .array()
      .of(
        yup.object().shape({
          permit_number: yup.string().max(100, 'Cannot exceed 100 characters').required('Required'),
          permit_type: yup.string().required('Required')
        })
      )
      .isUniquePermitNumber('Permit numbers must be unique')
  })
});

/**
 * Create project - Permit section
 *
 * @return {*}
 */
const ProjectPermitForm: React.FC = () => {
  const classes = useStyles();

  const { values, handleChange, getFieldMeta, errors } = useFormikContext<IProjectPermitForm>();

  const permitTypes = [
    'Forestry License to Cut',
    'Heritage Inspection Permit',
    'License of Occupation',
    'Occupant License to Cut under the Forest Act',
    'Park Use Permit',
    'Road Use Permit',
    'Special Use Permit'
  ];

  return (
    <>
      <FieldArray
        name="permit.permits"
        render={(arrayHelpers) => (
          <>
            <Box mb={3} maxWidth={'72ch'}>
              <Typography variant="body1" color="textSecondary">
                <strong>Note:</strong> For permit numbers, only provide the last 6 digits located after the hyphen (e.g.
                for KA12-845782 enter 845782)
              </Typography>
            </Box>
            {values.permit.permits?.map((permit, index) => {
              const permitNumberMeta = getFieldMeta(`permit.permits.[${index}].permit_number`);
              const permitTypeMeta = getFieldMeta(`permit.permits.[${index}].permit_type`);

              return (
                /* Permit List List */
                <Grid container spacing={3} key={index}>
                  <Grid item xs={12} md={6}>
                    <List>
                      <ListItem className={classes.customListItem}>
                        <Grid container spacing={3}>
                          <Grid item xs={6}>
                            <FormControl fullWidth variant="outlined">
                              <InputLabel id="permit-type-label">Permit Type</InputLabel>
                              <Select
                                id="permit-type-select"
                                name={`permit.permits.[${index}].permit_type`}
                                labelId="permit-type-label"
                                label="Permit Type"
                                value={permit.permit_type}
                                onChange={handleChange}
                                error={permitTypeMeta.touched && Boolean(permitTypeMeta.error)}
                                displayEmpty
                                inputProps={{ 'aria-label': 'Permit Type' }}>
                                {permitTypes.map((permitType, index) => (
                                  <MenuItem key={index} value={permitType}>
                                    {permitType}
                                  </MenuItem>
                                ))}
                              </Select>
                              <FormHelperText>{permitTypeMeta.touched && permitTypeMeta.error}</FormHelperText>
                            </FormControl>
                          </Grid>
                          <Grid item xs={6}>
                            <CustomTextField
                              name={`permit.permits.[${index}].permit_number`}
                              label="Permit Number"
                              other={{
                                required: true,
                                value: permit.permit_number,
                                error: permitNumberMeta.touched && Boolean(permitNumberMeta.error),
                                helperText: permitNumberMeta.touched && permitNumberMeta.error
                              }}
                            />
                          </Grid>
                        </Grid>
                        <ListItemSecondaryAction>
                          <IconButton
                            color="primary"
                            data-testid="delete-icon"
                            aria-label="remove permit"
                            onClick={() => arrayHelpers.remove(index)}
                            edge="end">
                            <Icon path={mdiTrashCanOutline} size={1}></Icon>
                          </IconButton>
                        </ListItemSecondaryAction>
                      </ListItem>
                    </List>
                  </Grid>
                </Grid>
              );
            })}
            <Box pt={2}>
              <Button
                type="button"
                variant="outlined"
                color="primary"
                aria-label="add permit"
                startIcon={<Icon path={mdiPlus} size={1}></Icon>}
                onClick={() => arrayHelpers.push(ProjectPermitFormArrayItemInitialValues)}>
                Add New Permit
              </Button>
            </Box>
          </>
        )}
      />
      <Box>
        {errors.permit?.permits && !Array.isArray(errors.permit?.permits) && (
          <Box pt={2}>
            <Typography style={{ fontSize: '12px', color: '#f44336' }}>{errors.permit.permits}</Typography>
          </Box>
        )}
      </Box>
    </>
  );
};

export default ProjectPermitForm;
