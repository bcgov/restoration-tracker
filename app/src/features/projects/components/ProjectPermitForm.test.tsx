import { fireEvent, render, waitFor } from '@testing-library/react';
import { Formik } from 'formik';
import React from 'react';
import ProjectPermitForm, {
  IProjectPermitForm,
  ProjectPermitFormInitialValues,
  ProjectPermitFormYupSchema
} from './ProjectPermitForm';

describe('ProjectPermitForm', () => {
  it('renders correctly with default empty values', () => {
    const { getByTestId } = render(
      <Formik
        initialValues={ProjectPermitFormInitialValues}
        validationSchema={ProjectPermitFormYupSchema}
        validateOnBlur={true}
        validateOnChange={false}
        onSubmit={async () => {}}>
        {() => <ProjectPermitForm />}
      </Formik>
    );

    expect(getByTestId('permit.permits.[0].permit_number')).toBeVisible();
  });

  it('renders correctly with existing permit values', () => {
    const existingFormValues: IProjectPermitForm = {
      permit: {
        permits: [
          {
            permit_number: '123',
            permit_type: 'Park Use Permit'
          },
          {
            permit_number: '3213123123',
            permit_type: 'Scientific Fish Collection Permit'
          }
        ]
      }
    };

    const { getByTestId, getByText } = render(
      <Formik
        initialValues={existingFormValues}
        validationSchema={ProjectPermitFormYupSchema}
        validateOnBlur={true}
        validateOnChange={false}
        onSubmit={async () => {}}>
        {() => <ProjectPermitForm />}
      </Formik>
    );

    expect(getByTestId('permit.permits.[0].permit_number')).toBeVisible();
    expect(getByText('Park Use Permit')).toBeVisible();
    expect(getByText('Scientific Fish Collection Permit')).toBeVisible();
  });

  it('deletes existing permits when delete icon is clicked', async () => {
    const existingFormValues: IProjectPermitForm = {
      permit: {
        permits: [
          {
            permit_number: '123',
            permit_type: 'Scientific Fish Collection Permit'
          }
        ]
      }
    };

    const { getByTestId, queryByText } = render(
      <Formik
        initialValues={existingFormValues}
        validationSchema={ProjectPermitFormYupSchema}
        validateOnBlur={true}
        validateOnChange={false}
        onSubmit={async () => {}}>
        {() => <ProjectPermitForm />}
      </Formik>
    );

    expect(queryByText('Permit Number')).toBeInTheDocument();

    fireEvent.click(getByTestId('delete-icon'));

    await waitFor(() => {
      expect(queryByText('Permit Number')).toBeNull();
    });
  });
});
