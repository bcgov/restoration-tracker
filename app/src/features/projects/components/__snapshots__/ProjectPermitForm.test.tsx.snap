// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPermitForm renders correctly with error on the permits field due to duplicates 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-38"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <strong>
        Note:
      </strong>
       For permit numbers, only provide the last 6 digits located after the hyphen (e.g. for KA12-845782 enter 845782)
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-container"
        >
          <div
            class="MuiListItem-root makeStyles-customListItem-36 MuiListItem-gutters MuiListItem-secondaryAction"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    id="permit-type-label"
                  >
                    Permit Type
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-label="Permit Type"
                      aria-labelledby="permit-type-label permit-type-select"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                      id="permit-type-select"
                      role="button"
                      tabindex="0"
                    >
                      Park Use Permit
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput"
                      name="permit.permits.[0].permit_type"
                      tabindex="-1"
                      value="Park Use Permit"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-41 PrivateNotchedOutline-legendNotched-42"
                      >
                        <span>
                          Permit Type
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    for="permit.permits.[0].permit_number"
                    id="permit.permits.[0].permit_number-label"
                  >
                    Permit Number
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="permit.permits.[0].permit_number"
                      name="permit.permits.[0].permit_number"
                      required=""
                      type="text"
                      value="123"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-41 PrivateNotchedOutline-legendNotched-42"
                      >
                        <span>
                          Permit Number *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="remove permit"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
              data-testid="delete-icon"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-container"
        >
          <div
            class="MuiListItem-root makeStyles-customListItem-36 MuiListItem-gutters MuiListItem-secondaryAction"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    id="permit-type-label"
                  >
                    Permit Type
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-label="Permit Type"
                      aria-labelledby="permit-type-label permit-type-select"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                      id="permit-type-select"
                      role="button"
                      tabindex="0"
                    >
                      Scientific Fish Collection Permit
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput"
                      name="permit.permits.[1].permit_type"
                      tabindex="-1"
                      value="Scientific Fish Collection Permit"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-41 PrivateNotchedOutline-legendNotched-42"
                      >
                        <span>
                          Permit Type
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    for="permit.permits.[1].permit_number"
                    id="permit.permits.[1].permit_number-label"
                  >
                    Permit Number
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="permit.permits.[1].permit_number"
                      name="permit.permits.[1].permit_number"
                      required=""
                      type="text"
                      value="123"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-41 PrivateNotchedOutline-legendNotched-42"
                      >
                        <span>
                          Permit Number *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="remove permit"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
              data-testid="delete-icon"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-43"
  >
    <button
      aria-label="add permit"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            role="presentation"
            style="width: 1.5rem; height: 1.5rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
              style="fill: currentColor;"
            />
          </svg>
        </span>
        Add New Permit
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root-44"
  />
</DocumentFragment>
`;

exports[`ProjectPermitForm renders correctly with errors on the permit_number and permit_type fields 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-27"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <strong>
        Note:
      </strong>
       For permit numbers, only provide the last 6 digits located after the hyphen (e.g. for KA12-845782 enter 845782)
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-container"
        >
          <div
            class="MuiListItem-root makeStyles-customListItem-25 MuiListItem-gutters MuiListItem-secondaryAction"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    id="permit-type-label"
                  >
                    Permit Type
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-label="Permit Type"
                      aria-labelledby="permit-type-label permit-type-select"
                      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                      id="permit-type-select"
                      role="button"
                      tabindex="0"
                    >
                      Scientific Fish Collection Permit
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput"
                      name="permit.permits.[0].permit_type"
                      tabindex="-1"
                      value="Scientific Fish Collection Permit"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-28 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-30 PrivateNotchedOutline-legendNotched-31"
                      >
                        <span>
                          Permit Type
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    for="permit.permits.[0].permit_number"
                    id="permit.permits.[0].permit_number-label"
                  >
                    Permit Number
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="permit.permits.[0].permit_number"
                      name="permit.permits.[0].permit_number"
                      required=""
                      type="text"
                      value="123"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-28 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-30 PrivateNotchedOutline-legendNotched-31"
                      >
                        <span>
                          Permit Number *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root"
          >
            <button
              aria-label="remove permit"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
              data-testid="delete-icon"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  role="presentation"
                  style="width: 1.5rem; height: 1.5rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-32"
  >
    <button
      aria-label="add permit"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            role="presentation"
            style="width: 1.5rem; height: 1.5rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
              style="fill: currentColor;"
            />
          </svg>
        </span>
        Add New Permit
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root-33"
  />
</DocumentFragment>
`;
